packages/compiler/core/checker.ts

```diff
@@ -935,44 +955,24 @@ export function createChecker(program: Program): Checker {
         if (symbolLinks.declaredType) {
           baseType = symbolLinks.declaredType;
         } else {
-          baseType =
-            sym.flags & SymbolFlags.Model
-              ? checkModelStatement(decl as ModelStatementNode, mapper)
-              : sym.flags & SymbolFlags.Scalar
-              ? checkScalar(decl as ScalarStatementNode, mapper)
-              : sym.flags & SymbolFlags.Alias
-              ? checkAlias(decl as AliasStatementNode, mapper)
-              : sym.flags & SymbolFlags.Interface
-              ? checkInterface(decl as InterfaceStatementNode, mapper)
-              : sym.flags & SymbolFlags.Operation
-              ? checkOperation(decl as OperationStatementNode, mapper)
-              : checkUnion(decl as UnionStatementNode, mapper);
+          baseType = checkDeclaredType(decl, sym, mapper);
```

Comment:
For review: I just refactored this into its own function as we had that reused exactly the same right below